token/services/identity/msp/idemix/bccsp.go

```diff
@@ -7,34 +7,82 @@ SPDX-License-Identifier: Apache-2.0
 package idemix
 
 import (
-	idemix2 "github.com/IBM/idemix/bccsp"
+	idemix "github.com/IBM/idemix/bccsp"
 	"github.com/IBM/idemix/bccsp/keystore"
+	idemix3 "github.com/IBM/idemix/bccsp/schemes/dlog/crypto"
+	"github.com/IBM/idemix/bccsp/schemes/dlog/crypto/translator/amcl"
 	bccsp "github.com/IBM/idemix/bccsp/types"
 	math "github.com/IBM/mathlib"
 	"github.com/pkg/errors"
 )
 
-// NewKVSBCCSP returns a new BCCSP for the passed curve, if the curve is BLS12_381_BBS, it returns the BCCSP implementation
-// based on aries.
-func NewKVSBCCSP(kvsStore keystore.KVS, curveID math.CurveID) (bccsp.BCCSP, error) {
-	if curveID == math.BLS12_381_BBS {
-		logger.Debugf("new aries KVS-based BCCSP")
-		return NewKSVBCCSP(kvsStore, curveID, true)
+func NewKeyStore(curveID math.CurveID, backend keystore.KVS) (bccsp.KeyStore, error) {
+	curve, tr, err := GetCurveAndTranslator(curveID)
+	if err != nil {
+		return nil, err
+	}
+	keyStore := &keystore.KVSStore{
+		Curve:      curve,
+		KVS:        backend,
+		Translator: tr,
 	}
-	logger.Debugf("new dlog KVS-based BCCSP")
-	return NewKSVBCCSP(kvsStore, curveID, false)
+	return keyStore, nil
 }
 
-// NewAriesBCCSP returns an instance of the idemix BCCSP for the given curve based on aries
-func NewAriesBCCSP() (bccsp.BCCSP, error) {
-	logger.Infof("new aries no-KeyStore BCCSP")
-	curve, tr, err := GetCurveAndTranslator(math.BLS12_381_BBS)
+// NewBCCSP returns an instance of the idemix BCCSP for the given curve and kvsStore
+func NewBCCSP(keyStore bccsp.KeyStore, curveID math.CurveID, aries bool) (bccsp.BCCSP, error) {
+	curve, tr, err := GetCurveAndTranslator(curveID)
 	if err != nil {
 		return nil, err
 	}
-	cryptoProvider, err := idemix2.NewAries(&keystore.Dummy{}, curve, tr, true)
+
+	var cryptoProvider bccsp.BCCSP
+	if aries {
+		cryptoProvider, err = idemix.NewAries(keyStore, curve, tr, true)
```

Comment:
Maybe a check for the curve in the case of Aries

Reply:
in reality aries now supports any curve